src/main.rs

```diff
@@ -40,10 +41,34 @@ struct StateRow {
     event_id: String,
 }
 
+fn parse_request_uri(req_uri: RequestUri) -> hyper::Url {
+    match req_uri {
+        RequestUri::AbsolutePath(s) => {
+            // ffs
+            return hyper::Url::parse(&format!("http://foo{}", s)).unwrap();
+        }
+        RequestUri::AbsoluteUri(s) => { return s; }
+        _ => { panic!("unsupported uri type"); }
+    }
+}
 
-fn room(params: Params, _: Request, mut res: Response) {
+fn room(params: Params, req: Request, mut res: Response) {
     let room_id = params.find("room_id").expect("room_id not in params");
 
+    /* please tell me there is an easier way to do this */
+    let uri = parse_request_uri(req.uri);
+    let qs: BTreeMap<_, _> = uri.query_pairs().collect();
+
+    let max_depth = match qs.get("max_depth") {
+        Some(x) => {
+            x.parse::<i64>().expect(
```

Comment:
Personally I prefer to instead annotate the variable declaration instead of specifying the parameters:

```rust
let max_depth: i64 = ...
```